docker-entrypoint.sh

```diff
@@ -35,10 +36,27 @@ fi
   --upscript=${KARP_UPSCRIPT} \
   --downscript=${KARP_DOWNSCRIPT} \
   --xparam=${KARP_SUBNET} \
-  ${KARP_EXTRA_FLAGS} &
+  ${KARP_EXTRA_FLAGS}
 
-# wait for the last process sent to background to finish (ucarp)
-wait $!
+demote_ucarp() {
+  kill -USR2 "$(pidof ucarp)"
+  sleep 1
+  $KARP_DOWNSCRIPT "$KARP_INTERFACE" "$KARP_VIRTUAL_IP" "$KARP_SUBNET"
+}
+
+do_healthcheck() {
+  [ "$(wget --no-check-certificate -q -O - "$KARP_HEALTHCHECK_URL")" = ok ]
```

Comment:
This `wget` returns a 401 error when being run in an existing kube-karp pod in my cluster, so it won't be "ok" ever.

```bash
/ # wget --no-check-certificate -q -O - https://localhost:6443/livez
wget: server returned error: HTTP/1.1 401 Unauthorized
```

Reply:
That's strange.  OpenShift allows this URI without authorization.

Reply:
I'm using RKE (Rancher Kubernetes Engine), so this is a nice implicit test case but a truly blocker one if it doesn't pass